languagetool-core/src/main/java/org/languagetool/RuleEntityResolver.java

```diff
@@ -26,22 +23,19 @@ public class RuleEntityResolver implements EntityResolver {
    */
   @Override
   public InputSource resolveEntity(String publicId, String systemId) throws SAXException, IOException {
-    if ((publicId != null && publicId.endsWith(".ent")) || systemId.endsWith(".ent")) {
-      URL entitiesUrl = new URL(xmlUrl, new URL(systemId).getPath());
-      if (Objects.equals(entitiesUrl.getProtocol(), "jar")) {
-        InputStream entitiesStream = JLanguageTool.getDataBroker().getAsStream(entitiesUrl.toString().split("!")[1]);
-        return new InputSource(entitiesStream);
-      } else {
-        return new InputSource(entitiesUrl.getPath());
-      }
+    if (systemId != null && systemId.endsWith(".ent")) {
+      return new InputSource(getInputStreamLTEntities(systemId));
     }
     return null;
   }
 
-  /**
-   * @param xmlUrl path to the source XML where the external entity relative path is found.
-   */
-  public RuleEntityResolver(URL xmlUrl) {
-    this.xmlUrl = xmlUrl;
+  public String getPathFromLTResourceFolder(String input) {
+    // we assume that the entities file is in the resource folder of each language
```

Comment:
This is certainly an assumption, but what we're doing here isn't really resolving any paths relative to anything else, we're just hardcoding their location, and keeping the relative paths there only so XML editors can resolve them.

This is *fine*, but we're moving from a solution that actually resolved the paths as *relative* paths to one that *assumes* stuff about their location, so we should probably document this well.